### DISCUSSION TITLE
How to create a set of independently calculated quantum catch color data?

### DISCUSSION THREAD

**Discussion by yybearhao** [thomased/colRverse#10] [Q&A] [answered]
“vismodeldata” required quantum catch color data, can be either the result from [vismodel()](https://pavo.colrverse.com/reference/vismodel.html) or independently calculated data (in the form of a data frame with columns representing quantum catches).
My question is How to create a set of independently calculated quantum catch color data?

I got a set of quantum catch color data through ImageJ,
When I create a data frame with columns representing quantum catches, 
The system keeps displaying an error

> G1_2 <- read.csv("G1-2.csv")
> head(G1_2)
           s         m         l
1 0.07329026 0.1740861 0.3302621
2 0.02419910 0.1336217 0.2952741
3 0.08881478 0.1940379 0.3744964
4 0.08517226 0.3338720 0.6265567
5 0.10669659 0.2286708 0.3816763
6 0.05971873 0.2449469 0.4902438
> hex.G1_2 <- colspace(G1_2, space = "hexagon", qcatch = "Qi")
Warning message:
Input data is not a “vismodel” object; treating columns as quantum catch for “s”, “m”, and “l” receptors, respectively 

 Warning message:
Input data is not a “vismodel” object; treating columns as quantum catch for “s”, “m”, and “l” receptors, respectively

**Comment by Bisaloo** [comment, marked as answer]
Hi @yybearhao, thanks for getting in touch!

Your approach is perfectly correct. You get [a warning message, but not an error message](https://adv-r.hadley.nz/conditions.html#signalling-conditions), which means the computation successfully completed. The warning is here to inform the user they are slightly deviating from the usual pavo workflow and to ensure this is what they intend.

So, no worries, you can carry on with your analysis using `hex.G1_2`.

However, we are always keen on getting feedback to improve pavo. Do you have a suggestion for a better way to signal this?

**Reply by yybearhao** [reply]
Thank you very much for the reply, that's a great help to me.

I was stuck with the warning message and couldn't continue with the data analysis, which is now resolved.

Pavo is a very useful R package for me, I will continue learn and use it.
I f I have any suggestions or problems, I will contact you in time.
